### ISSUE TITLE
bug(core): createMicroservice without config throw TypeError: Cannot match against 'undefined' or 'null'.

### ISSUE THREAD

**Issue by cojack** [nestjs/nest#185]
Hi,
as from [docs](https://docs.nestjs.com/microservices/basics):

```typescript
import { NestFactory } from '@nestjs/core';
import { ApplicationModule } from './modules/app.module';
import { Transport } from '@nestjs/microservices';

async function bootstrap() {
  const app = await NestFactory.createMicroservice(ApplicationModule, {
    transport: Transport.TCP,
  });
  app.listen(() => console.log('Microservice is listening'));
}
bootstrap();
```

Remove the config from `createMicroservice` because he's optional.

```typescript
import { NestFactory } from '@nestjs/core';
import { ApplicationModule } from './modules/app.module';
import { Transport } from '@nestjs/microservices';

async function bootstrap() {
  const app = await NestFactory.createMicroservice(ApplicationModule);
  app.listen(() => console.log('Microservice is listening'));
}
bootstrap();
```

And if you will handle `.catch` on `bootstrap` function you will get an error:

`TypeError: Cannot match against 'undefined' or 'null'.`

If you will not handle `.catch` then you will find in console:

```
(node:12307) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot match against 'undefined' or 'null'.
(node:12307) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Version: "@nestjs/core@4.1.1"

**Comment by kamilmysliwiec** [comment]
Hi @cojack,
It's a bug, thanks. Will be fixed with the incoming update.

**Comment by kamilmysliwiec** [comment]
Fixed (v4.1.7), thanks again 🙂
